src/main/java/com/dev/theater/dao/impl/DaoImpl.java

```diff
@@ -2,17 +2,27 @@
 
 import com.dev.theater.dao.GenericDao;
 import com.dev.theater.exception.CrudException;
-import com.dev.theater.util.HibernateUtil;
 import org.hibernate.Session;
+import org.hibernate.SessionFactory;
 import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
 
+@Repository
 public class DaoImpl<T> implements GenericDao<T> {
+    protected final SessionFactory sessionFactory;
+
+    @Autowired
+    public DaoImpl(SessionFactory sessionFactory) {
```

Comment:
let's make this class abstract and see the following article
https://www.baeldung.com/spring-autowired-abstract-class#constructor-injection